app/models/merchant.rb

```diff
@@ -1,5 +1,8 @@
 class Merchant < ApplicationRecord
   has_many :items
+  validates :name, presence: true
+
+  enum status: [:disabled, :enabled]
```

Comment:
Did you try this as a hash by chance?  I tried it on mine and it didn't seem to work for some reason!  I have to switch it to an array format like yours.